### DISCUSSION TITLE
Help with Layout and SelectionPrompt<T>

### DISCUSSION THREAD

**Discussion by rynsp8** [spectreconsole/spectre.console#1192] [Q&A]
I'm having difficulty displaying a selectionPrompt to a layout and the message I'm getting makes me think it isn't possible. 

"Argument 1: cannot convert from 'Spectre.Console.SelectionPrompt<string>' to 'Spectre.Console.Rendering.IRenderable'"

My code is below.

`

    public class Program
    {

    static void Main()
    {
        //Start the application with a splash screen and menu options

        var layout = new Layout("Root")
            .SplitRows(
                new Layout("Top")
                .SplitRows(
                    new Layout("rootTitle"),
                    new Layout("rootCreator")),
                new Layout("Bottom")
            );
        //Raccoon Tycoon title
        var rootTitle = new Panel(new FigletText("Raccoon Tycoon!")
            .Color(Color.Red).Centered());
        rootTitle.Border = BoxBorder.None;
        rootTitle.Expand = false;

        //Raccoon Tycoon creator
        var rootAuthor = new Panel(
            Align.Center(
                new Markup("[green]by Glenn Drover[/]"))
            );
        rootAuthor.Border = BoxBorder.None;
        rootAuthor.Expand = false;

        //Raccoon Tycoon start menu
        var startMenu = new SelectionPrompt<string>()
        {
            Title = "Menu",
            PageSize = 2

        };
        
        layout["rootTitle"].Update(
            rootTitle
            ).Ratio(1);
        layout["rootCreator"].Update(
            rootAuthor
            ).Ratio(1);
        layout["bottom"].Update(
            startMenu
            );

        AnsiConsole.Write(layout);
    }`

**Comment by patriksvensson** [comment]
@rynsp8 That is by design. Interactive things such as prompts, live-view, etc cannot be placed inside other widgets.

**Reply by JamesNeedsChrist** [reply]
I have seen some programmers manage to put them inside the layout, which shocked me
